contrib/golang/filters/http/source/go/pkg/http/filter.go

```diff
@@ -99,5 +113,37 @@ type streamInfo struct {
 }
 
 func (s *streamInfo) GetRouteName() string {
-	return cAPI.HttpGetRouteName(unsafe.Pointer(s.request.req))
+	name, _ := cAPI.HttpGetStringValue(unsafe.Pointer(s.request.req), ValueRouteName)
+	return name
+}
+
+func (s *streamInfo) FilterChainName() string {
+	name, _ := cAPI.HttpGetStringValue(unsafe.Pointer(s.request.req), ValueFilterChainName)
+	return name
+}
+
+func (s *streamInfo) Protocol() (string, bool) {
+	if protocol, ok := cAPI.HttpGetIntegerValue(unsafe.Pointer(s.request.req), ValueProtocol); ok {
+		if name, ok := protocolsIdToName[protocol]; ok {
+			return name, true
+		}
+		panic(fmt.Sprintf("invalid protocol id: %d", protocol))
```

Comment:
invalid protocol id is not a fatal error, so it's better to return `"", false` directly.

Reply:
`Http::Protocol` is an enum class, there won't be other values.
https://github.com/envoyproxy/envoy/blob/main/envoy/http/protocol.h#L12